routing/rt/graph/path.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2016 Bitfury Group Limited
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php
+
+package graph
+
+import "errors"
+
+func Path(id1, id2 ID, parent map[ID]ID) ([]ID, error) {
```

Comment:
ReversePath(id1, id2, parent) returns path from id2 to id1.
Then we reverse it to obtain path from id1 to id2. It gets reversed only once.